tests/uptane_vector_tests.cc

```diff
@@ -45,67 +45,100 @@ class Uptane_Vector_Test {
-      Uptane::EcuSerial ecu_serial(config.provision.primary_ecu_serial);
-      Uptane::HardwareIdentifier hw_id(config.provision.primary_ecu_hardware_id);
-      uptane_client->hw_ids.insert(std::make_pair(ecu_serial, hw_id));
-      uptane_client->installed_images[ecu_serial] = "test_filename";
-      if (!uptane_client->uptaneIteration()) {
-        throw uptane_client->getLastException();
+    HttpClient http_client;
+    while (true) {
+      HttpResponse response = http_client.post(address + test_name + "/step", Json::Value());
+      if (response.http_status_code == 204) {
+        return true;
       }
+      const auto vector(response.getJson());
+      std::cout << "VECTOR: " << vector;
 
-    } catch (const Uptane::Exception& e) {
-      return match_error(vector, e);
-    } catch (const std::exception& e) {
-      std::cout << "aktualizr failed with unrecognized exception " << typeid(e).name() << ": " << e.what() << "\n";
-      return false;
-    }
+      bool should_fail = false;
+      if (vector["director"]["update"]["is_success"].asBool() == false ||
+          vector["image_repo"]["update"]["is_success"].asBool() == false) {
+        should_fail = true;
+      } else {
+        for (const auto& t : vector["director"]["targets"]) {
+          if (t["is_success"].asBool() == false) {
+            should_fail = true;
+            break;
+          }
+        }
+        for (const auto& t : vector["image_repo"]["targets"]) {
+          if (t["is_success"].asBool() == false) {
+            should_fail = true;
+            break;
+          }
+        }
+      }
 
-    if (vector["director"]["update"]["is_success"].asBool() && vector["image_repo"]["update"]["is_success"].asBool()) {
-      for (Json::ValueConstIterator it = vector["director"]["targets"].begin();
-           it != vector["director"]["targets"].end(); ++it) {
-        if (!(*it)["is_success"].asBool()) {
-          std::cout << "aktualizr did not fail as expected.\n";
+      try {
+        if (!uptane_client->uptaneIteration()) {
+          if (should_fail) {
+            throw uptane_client->getLastException();
+          }
+          std::cout << "aktualizr return error but should not\n";
           return false;
         }
-      }
-      for (Json::ValueConstIterator it = vector["image_repo"]["targets"].begin();
-           it != vector["image_repo"]["targets"].end(); ++it) {
-        if (!(*it)["is_success"].asBool()) {
-          std::cout << "aktualizr did not fail as expected.\n";
+        std::vector<Uptane::Target> updates;
+        if (!uptane_client->uptaneOfflineIteration(&updates, nullptr)) {
+          std::cout << "uptaneOfflineIteration returns false\n";
           return false;
+        } else {
+          if (updates.size()) {
+            if (!uptane_client->downloadImages(updates)) {
+              if (should_fail) {
+                throw uptane_client->getLastException();
+              }
```

Comment:
There should be another check here, right? If `downloadImages` failed but we didn't expect it, we should say something.